cudamapper/src/utils.cpp

```diff
@@ -305,7 +304,11 @@ void print_sam(const std::vector<Overlap>& overlaps,
         // TODO: write AUX data if available
 
         std::lock_guard<std::mutex> lg(write_output_mutex);
-        int out = sam_write1(file.get(), header.get(), alignment.get());
+        result = sam_write1(file.get(), header.get(), alignment.get());
+        if (result < 0)
+        {
+            fprintf(stderr, "print_sam: could not write alignment");
```

Comment:
Should we do something more here? Like throw an exception or similar

Reply:
That might be a good idea. `sam_write1()` has a couple of failure modes because internally it does some thread creation, (potentially) memory allocations, and then some io. Should there be different exceptions for each or just a generic runtime_error?

Reply:
I think a generic exception would suffice, one can then investigate further if needed. I just think it's important to signal the failure to the user in a way better than just a print

Reply:
👍 fixed in latest commit